proposals/csharp-9.0/records.md

```diff
@@ -197,6 +197,115 @@ If the containing record is abstract, the synthesized clone method is also abstr
 If the "clone" method is not abstract, it returns the result of a call to a copy constructor. 
 
 
+### Printing members: print and ToString methods
+
+If the record is derived from `object`, the record includes a synthesized "print" method equivalent to a method declared as follows, but with a compiler-reserved name:
+```C#
+void print(System.StringBuilder builder, bool includeSeparator);
+```
+The property is `private` if the record type is `sealed`. Otherwise, the property is `virtual` and `protected`.
+
+The method:
+1. appends a separator ", " to `builder` if `includeSeparator` is true and the record has public fields or properties,
+2. for each of the record's public field and property members, appends that member's name followed by " = " followed by the member's value: `this.member`, separated with ", ",
+
+If the record type is derived from a base record `Base`, the record includes a synthesized override equivalent to a "print" method declared as follows:
+```C#
+protected override void print(StringBuilder builder, bool includeSeparator);
+```
+
+The method appends the same contents to `builder` as described above, but also then calls the base "print" method with two arguments:
+1. its `builder` parameter,
+2. a true `includeSeparator` argument if either (a) the record has public fields or properties, or (b) its `includeSeparator` parameter was given as true.
+
+The record includes a synthesized method equivalent to a method declared as follows:
+```C#
+public override string ToString();
+```
+
+The method can be declared explicitly. It is an error if the explicit declaration does not match the expected signature or accessibility, or if the explicit declaration doesn't allow overiding it in a derived type and the record type is not `sealed`. It is an error if the method doesn't override a method with this signature in the record `Base` (for example, if the method is missing in the `Base`, or sealed, or not virtual, etc.).
```

Comment:
>It is an error if the method doesn't override a method with this signature in the record `Base` (for example, if the method is missing in the `Base`, or sealed, or not virtual, etc.). [](start = 251, length = 183)

I think this requirement should be stronger, i.e. similar to what we have for GetHashCode, for example. Specifically, something like this: "It is an error if either synthesized, or explicitly declared method doesn't override `object.ToString()` (for example, due to shadowing in intermediate base types, etc.)."